File: android/app/src/main/java/com/devnokiyo/actioncableapp/enums/SocketType.kt
package com.devnokiyo.actioncableapp.enums

import com.google.gson.annotations.SerializedName


enum class SocketType(val rawValue: String) {
    @SerializedName("in")
    RoomIn("in"),
    @SerializedName("out")
    RoomOut("out"),
    @SerializedName("mumbling")
    Mumbling("mumbling"),
    @SerializedName("bark")
    Bark("bark")
}

File: ios/ActionCableApp/Enums/SocketType.swift
enum SocketType: String, Codable {
    case roomIn = "in"
    case roomOut = "out"
    case mumbling = "mumbling"
    case bark = "bark"
    
    init?(type: String) {
        switch type {
        case SocketType.roomIn.rawValue:
            self = .roomIn
        case SocketType.roomOut.rawValue:
            self = .roomOut
        case SocketType.mumbling.rawValue:
            self = .mumbling
        case SocketType.bark.rawValue:
            self = .bark
        default:
            return nil
        }
    }
}
